sql/moz-fx-data-shared-prod/telemetry_derived/firefox_android_clients_v1/query.sql

```diff
@@ -0,0 +1,482 @@
+-- Query first observations for Firefox Android Clients.
+-- Proposal:
+-- https://docs.google.com/document/d/12bj4DhCybelqHVgOVq8KJlzgtbbUw3f68palNrv-gaM/edit#
+CREATE TEMP FUNCTION get_first_not_null_value(
+  existing_value STRING,
+  existing_date DATE,
+  first_session_value STRING,
+  first_session_date DATE,
+  metrics_value STRING,
+  metrics_date DATE
+) AS (
+  -- Return first not null value, or null.
+  CASE
+  WHEN
+    existing_value IS NULL
+  THEN
+    COALESCE(first_session_value, metrics_value)
+  WHEN
+    existing_value IS NOT NULL
+    AND first_session_value IS NOT NULL
+    AND first_session_date < existing_date
+  THEN
+    first_session_value
+  WHEN
+    existing_value IS NOT NULL
+    AND first_session_value IS NULL
+    AND metrics_value IS NOT NULL
+    AND metrics_date < existing_date
+  THEN
+    metrics_value
+  ELSE
+    existing_value
+  END
+);
+
+CREATE TEMP FUNCTION get_first_not_null_date(
```

Comment:
We can have the above function also return the `DATE` so the single function can serve both purposes

Reply:
For ref. This logic is moved to mozfun.norm.get_earliest_value.